### PR TITLE
Update cdi-builder to use go version 1.14.6 (current latest)

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Fix functional test framework to work with go >= 1.13. We were not
using the ginkgo ordering correctly, and our framework would call
flag.Parse() too soon, and mess with the go testing flag parsing.
As a result we had to alter some of the variables that we use to build
urls to be functions so that when called the required information is
available.
Fixed several tests not checking for errors.
Fixed dumping code being initialized several times messing with the
order in which failures were reported.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Enhancement: CDI now uses golang 1.14.6
```

### CODE REVIEW

File: tests/tests_suite_test.go [hunk 1]
```diff
@@ -1,17 +1,84 @@
...
 )
 
 func TestTests(t *testing.T) {
 	defer GinkgoRecover()
 	RegisterFailHandler(tests.CDIFailHandler)
+	BuildTestSuite()
 	RunSpecsWithDefaultAndCustomReporters(t, "Tests Suite", reporters.NewReporters())
 }
+
+// To understand the order in which things are run, read http://onsi.github.io/ginkgo/#understanding-ginkgos-lifecycle
```
Comment (brybacki) [inline tests/tests_suite_test.go:35]: good to have this info

File: hack/build/docker/builder/Dockerfile [hunk 2]
```diff
@@ -42,7 +42,7 @@ RUN pip3 install j2cli && pip3 install operator-courier && \
     ln -s /opt/gradle/gradle-4.3.1/bin/gradle /usr/local/bin/gradle && \
     rm gradle-4.3.1-bin.zip
 
-ENV GIMME_GO_VERSION=1.12.14 GOPATH="/go" KUBEBUILDER_VERSION="2.2.0" ARCH="amd64"
+ENV GIMME_GO_VERSION=1.14.6 GOPATH="/go" KUBEBUILDER_VERSION="2.3.1" ARCH="amd64"
```
Comment (mhenriks) [inline hack/build/docker/builder/Dockerfile:45]: Should these changes be separate (and committed before) from the rest?

File: tests/framework/framework.go [hunk 3]
```diff
@@ -96,152 +98,34 @@ type Framework struct {
...
-	utils.CacheTestsData(f.K8sClient, f.CdiInstallNs)
-	f.mux.Unlock()
-
-	return f, err
+	return f
 }
 
 // BeforeEach provides a set of operations to run before each test
 func (f *Framework) BeforeEach() {
-	if !f.SkipControllerPodLookup {
```
Comment (brybacki) [inline tests/framework/framework.go:236]: not needed anymore? I cannot find it

File: tests/datavolume_test.go [hunk 4]
```diff
@@ -763,18 +790,19 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
...
 					dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "1Gi", url)
 
 					By(fmt.Sprintf("creating new datavolume %s", dataVolume.Name))
 					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
 					Expect(err).ToNot(HaveOccurred())
 
 					By(fmt.Sprintf("waiting for datavolume to match phase %s", cdiv1.Succeeded))
-					utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, dataVolume.Name)
+					err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, dataVolume.Name)
+					Expect(err).ToNot(HaveOccurred())
```
Comment (brybacki) [inline tests/datavolume_test.go:805]: good catch, 
I think I know why it happens, some of our test utils do check and fail inside the function logic, and some are just returning an err value so the caller can assert it,

File: hack/build/run-functional-tests.sh [hunk 5]
```diff
@@ -72,6 +72,5 @@ if [ $retry_counter -eq $MAX_CDI_WAIT_RETRY ]; then
     exit 1
 fi
 
-test_command="${TESTS_OUT_DIR}/tests.test -test.timeout 180m ${test_args}"
-echo "${test_command}"
+test_command="${TESTS_OUT_DIR}/tests.test -test.timeout 360m ${test_args}"
```
Comment (brybacki) [inline hack/build/run-functional-tests.sh:75]: why 360m?

File: tests/framework/framework.go (hunk 3 above)
Comment (awels) [reply tests/framework/framework.go:236]: It was never used, so I figured I might as well remove it while I was cleaning house.

File: tests/datavolume_test.go (hunk 4 above)
Comment (awels) [reply tests/datavolume_test.go:805]: Yeah, well I noticed this should have failed while I was testing and the test 'passed' so now it should fail properly if it fails.

File: hack/build/run-functional-tests.sh (hunk 5 above)
Comment (awels) [reply hack/build/run-functional-tests.sh:75]: With some failures 180 was not long enough any more. so I increased it

File: hack/build/docker/builder/Dockerfile (hunk 2 above)
Comment (awels) [reply hack/build/docker/builder/Dockerfile:45]: They can, however this PR should work with both go 12.14 and 14.6. I am going to need a follow up PR to change the container hash regardless.

File: tests/framework/framework.go (hunk 3 above)
Comment (awels) [reply tests/framework/framework.go:236]: So I was wrong, I found a location it is used.

**Pull request merged**